app/scripts/models/pairing/supplicant-state-machine.js

```diff
@@ -0,0 +1,161 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import PairingChannelClientErrors from '../../lib/pairing-channel-client-errors';
+import { CompleteState, State } from './state';
+import PairingFlowStateMachine from './state-machine';
+
+/* eslint-disable no-use-before-define */
+
+class SupplicantState extends State {
+  constructor (attributes, options = {}) {
+    super(attributes, options);
+
+    this.pairingChannelClient = options.pairingChannelClient;
+    this.listenTo(this.pairingChannelClient, 'close', () => this.socketClosed());
+    this.listenTo(this.pairingChannelClient, 'error', (error) => this.socketError(error));
+  }
+
+  socketClosed () {
+    this.navigate('pair/failure', {
+      error: PairingChannelClientErrors.toError('CONNECTION_CLOSED')
+    });
+  }
+
+  socketError (error) {
+    this.navigate('pair/failure', { error });
```

Comment:
Should this error be logged to Sentry or will that just lead to unnecessary chatter? Will it be logged in the `pair/failure` view?

Reply:
👍

Reply:
I will add it as a follow up, need to check what it logs and also will require more code changes